### ISSUE TITLE
Gitlog fails when the subject or body includes quotation marks

### ISSUE THREAD

**Issue by am** [rubenv/grunt-git#101]
When running the task it fails with the following error:
```
Warning: Log failed to match exepected pattern:
{
  "hash": "60dc76bd83bfaf1a9380683369398863847d51e6",
  "author": {
    "name": "foo",
    "email": "bla@bla.io"
  },
  "date": "Tue, 10 Feb 2015 10:12:42 +0100",
  "subject": "Merge pull request #705 from io/joe/123",
  "body": "Fixes boo "word" bar"
}
```

Is there any way to escape the values before parsing the json?

**Comment by nburana** [comment]
Having the same issue.  In the meantime is there a way that client code can specify the format to remove subject and body?

**Comment by nburana** [comment]
In case this helps anyone, you can set the format in the config file to exclude subject and body (assuming you dont need it) and it wont break the json parser.

module.exports = function (options) {
  return {
    head: {
      options: {
        prop: 'gitlog.head.result',
        number: 1,
        pretty: 'format:{%n' +
          '  "hash": "%H",%n' + // commit hash
          '  "author": {%n' +
          '    "name": "%an",%n' + // author
          '    "email": "%ae"%n' + // email
          '  },%n' +
          '  "date": "%aD"%n' + // date
          '}%n' +
          '--grunt-gitlog-separator--'
      }
    }
  }
}

**Comment by am** [comment]
@nburana the problem persists if you need that field...

**Comment by nburana** [comment]
Yeah totally.  We just needed the hash and the date so this worked for us.

**Comment by am** [comment]
Looking into the source-code it will be difficult to fix this if we don't do a more granular parsing of the git log output. 
When logging a range of commits the [handleResult](https://github.com/rubenv/grunt-git/blob/master/lib/command_log.js#L107) method will receive a string that almost have the complete format of a JSON. This is where the problem starts, since it's now too late to redefine the parsing.
Looking into a similar [question in SO](http://stackoverflow.com/questions/9301673/can-i-escape-chars-in-git-log-output) the [solution](http://stackoverflow.com/a/9377663) implies that we parse each commit individually, even more, for each field we want to get there is a `git log` for that:
```shell
function escape_chars {
    sed -r 's/(\{\}")/\\\1/g'
}
function format {
    sha=$(git log -n1 --pretty=format:%h $1 | escape_chars)
    message=$(git log -n1 --pretty=format:%B $1 | escape_chars)
    author=$(git log -n1 --pretty=format:'%aN <%aE>' $1 | escape_chars)
    commit=$(git log -n1 --pretty=format:%cE $1 | escape_chars)
    date=$(git log -n1 --pretty=format:%cD $1 | escape_chars)
    echo "{\"sha\":\"$sha\",\"message\":\"$message\",\"author\":\"$author\",\"commit\":\"$commit\",\"date\":\"$date\"}"
}

for hash in $(git rev-list)
do
  format $hash
done
```
Not sure if this would be too much overhead for this task. 
Any thoughts?

**Comment by dbrill** [comment]
Has anyone found a reasonable work around for this. 

Would anyone benefit from me submitting a PR. If this comment gets a reasonable amount of notice I'll submit something.
